lib/oneview-sdk/resource/api200/lig_uplink_set.rb

```diff
@@ -40,15 +40,24 @@ def add_network(network)
 
       # Specify one uplink passing the virtual connect bay and the port to be attached.
       # @param [Fixnum] bay number to identify the VC
-      # @param [String] port to attach the uplink. Allowed D1..D16 and X1..X10
-      def add_uplink(bay, port)
+      # @param [String, Fixnum] port to attach the uplink. Examples: X1, D1, Q1, Q1.1, Q1:1, 67 ...
+      # @param [String] interconnect model name
+      # @param [Fixnum] enclosure number for multi-frame configurations
+      def add_uplink(bay, port, type = nil, enclosure_index = 1)
+        enclosure_index = !type.nil? && type.include?('Virtual Connect SE 16Gb FC Module') ? -1 : enclosure_index
+        port =
+          if type
+            fetch_relative_value_of(port, type)
+          else
+            port.to_s == port.to_i.to_s ? port : relative_value_of(port)
```

Comment:
Nope, it's fine as is! :smile: 

I would just add a comment explaining it, since it's not quite clear at first sight.